doc/kuksa_requirementes.md

```diff
@@ -0,0 +1,787 @@
+  Error error                     = 1;
+}
+
+message EntryUpdate {
+  DataEntry entry                 = 1;
+  repeated Field fields           = 2;
+}
+
+message SetRequest {
+  repeated EntryUpdate updates    = 1;
+}
+
+message SetResponse {
+  Error error                     = 1;
+}
+
+message SubscribeEntry {
+  string path                     = 1;
+  View view                       = 2;
+  repeated Field fields           = 3;
+}
+
+message SubscribeRequest {
+  repeated SubscribeEntry entries = 1;
+}
+
+message SubscribeResponse {
+  repeated EntryUpdate updates    = 1;
+}
+
+message ActuateRequest {
+  EntryUpdate actuator = 1;
+}
+
+message ActuateResponse {
+  Error error                     = 1;
+}
+
+message MetadataRequest {
+  repeated string path            = 1;
+}
+
+message MetadataResponse {
+  repeated DataEntry metadata     = 1;
+}
+
+message GetServerInfoRequest {
+  // Nothing yet
+}
+
+message GetServerInfoResponse {
+  string name                     = 1;
+  string version                  = 2;
+}
+``` 
+
+# Functional requirements
+
+### As Client
+1. As a client I want to have a single service point accessible from my execution environment.
+   
+2. As a client I want to `Get` sensor values or actuator current values of the vehicle and its environment in a uniform manner.
+  - Details:
+    * All sensor and actuator values for which a client have permission to.
+    * Client must provide the path, paths or wildcard.
+    * I want to receive exactly all the signals’ values I requested without errors.
+  
+3. As a client I want to `Subscribe` to sensor values or actuator current values when some signal value changed -> what about continuous subscription?
+  - Details:
+    * All sensor and actuator values for which a client have permission to.
+    * Client must provide the path, paths or wildcard paths.
+    * I want to receive exactly all the signals’ values I requested without errors.
+
+4. As a client I want to `Subscribe` and be able to apply a filter to receive a sensor value with an interval of x milliseconds only if the value has changed.
+   
+5.  As a client I want to `GetMetadata` static Metadata from signals.
+  - Details:
+    * All sensors, actuators, and attributes values for which a client have permission to.
+    * Client must provide the path, paths or wildcard.
+
+6.  As a client I want to `Get` attributes values once.
+
+7.  As a client I want to `Actuate` a target value of an actuator and guarantee that the operation succeeded.
+
+8.  As a client I want to connect and interact with Databroker either if it contains data or not.
+
+9.  As a client I want to connect and interact with Databroker either if it contains data or not.
+
+10.  As a client I want to know if there is a provider up and running by getting some sensor or actuator value.
+
+11.  As a client I don’t want to `Subscribe` to target values.
+
+12.  As a client I don’t want to `Subscribe` to attributes values.
```

Comment:
Not yet fully sure about that: What about updating individual components on a running system (without restarting "everything")? In that case certain attributes may change and clients need being notified about that.

Reply:
I am not sure either whether we need this explicit exclusion. However, the VSS documentation (https://covesa.github.io/vehicle_signal_specification/rule_set/data_entry/attributes/) states: 

`Attribute values can also change, similar to sensor values. The latter can be useful for attribute values that are likely to change during the lifetime of the vehicle. However, attribute values should typically not change more than once per ignition cycle, or else it should be defined as a sensor instead.`

This definition would match with the assumption that attributes (should) do not change during an ignition cycle. But do clients "survive" an ignition cycle?